mods/prevent-focus-stealing.wh.cpp

```diff
@@ -0,0 +1,371 @@
+            } else if (wcschr(exe_setting, L'\\') != nullptr) {
+                is_match = _wcsnicmp(exe_path, exe_setting, size) == 0 && exe_path[size] == 0;
+            } else {
+                is_match = wcslen(exe_name) == size && _wcsnicmp(exe_name, exe_setting, size) == 0;
+            }
+            if (is_match) {
+                if (!process_args) {
+                    process_args = GetCommandLineW();
+                    //Wh_Log(L"Process command line: %ls", process_args);
+                    if (*process_args == L'"') {
+                        process_args++;
+                        while (*process_args && *process_args != L'"')
+                            process_args++;
+                        if (*process_args == L'"')
+                            process_args++;
+                    } else {
+                        while (*process_args && *process_args != L' ' && *process_args != L'\t')
+                            process_args++;
+                    }
+                    while (*process_args == L' ' || *process_args == L'\t')
+                        process_args++;
+                }
+                //Wh_Log(L"Process arguments: %ls", process_args);
+                for (int arg_index = 0; ; arg_index++) {
+                    PCWSTR arg = Wh_GetStringSetting(L"applications[%d].args[%d]", application_index, arg_index);
+                    if (!*arg) {
+                        Wh_FreeStringSetting(arg);
+                        break;
+                    }
+                    if (wcsstr(process_args, arg) == nullptr)
+                        is_match = false;
+                    Wh_FreeStringSetting(arg);
+                }
+                for (int arg_index = 0; ; arg_index++) {
+                    PCWSTR arg = Wh_GetStringSetting(L"applications[%d].exclude_args[%d]", application_index, arg_index);
+                    if (!*arg) {
+                        Wh_FreeStringSetting(arg);
+                        break;
+                    }
+                    if (wcsstr(process_args, arg) != nullptr)
+                        is_match = false;
+                    Wh_FreeStringSetting(arg);
+                }
+                if (is_match) {
+                    PCWSTR title = Wh_GetStringSetting(L"applications[%d].title", application_index);
+                    bool never_focus = Wh_GetIntSetting(L"applications[%d].never_focus", application_index);
+                    Wh_Log(L"Hooking process for window title: %ls", title);
+                    g_includedWindows.push_back(WindowInfo { title, never_focus });
+                    break;
+                }
+            }
+            Wh_FreeStringSetting(exe_setting);
+        }
+    }
+Done:
+    if (g_includedWindows.empty()) {
+        return FALSE;
+    }
+
+    if (!Wh_SetFunctionHook((void *)CreateWindowExW, (void *)CreateWindowExW_Hook, (void **)&CreateWindowExW_Orig)) {
+        Wh_Log(L"Failed to hook CreateWindowExW");
+        return FALSE;
```

Comment:
Note that in this case, `Wh_FreeStringSetting` isn't called for allocated data.

You might want to consider using 9`WindhawkUtils::StringSetting`](https://github.com/ramensoftware/windhawk/wiki/Development-tips#notable-utilities) which calls `Wh_FreeStringSetting` on destruction.